ext/opcache/ZendAccelerator.c

```diff
@@ -2064,6 +2064,8 @@ zend_op_array *persistent_compile_file(zend_file_handle *file_handle, int type)
 		    }
 
 			if (file_handle->opened_path) {
+				zend_shared_alloc_lock();
+
```

Comment:
Please think about a different solution. This one should make a big slowdown.

opcache should be able to do lock-free cache lookups.
Probably, the modification of  ``ZCSG(hash)`` should be done more careful (to keep consistency for concurrent readers)